### ISSUE TITLE
[Findability] User can't determine next steps in a flow.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#75981]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** govcio-vfep
**Product:** GI Bill Comparison Tool
**Feature:** Specialized Mission Filter

## Findings details
**VA.gov Experience Standard - issue:** User can't determine next steps in a flow.
**VA.gov Experience Standard - category:** Findability
**Launch-blocking:** Yes
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When a user attempts to apply filters without first entering a search term, an error appears ("Please fill in a school, employer, or training provider.") However, keyboard focus doesn't move back up to the error message --- meaning keyboard users will have to tab back through 25 tab stops to get back to where they need to enter their search term. 

And that assumes they realize that there's an error at all, regardless of if they're keyboard-only users. If they've scrolled far enough down the page that they can't see the error message when it appears, they may assume the "Apply filters" button just didn't work at all.

For screen reader users, the error message does get announced on its first occurence, so they'll hear what's wrong. But the expected pattern on VA.gov would be for their next tab stop to take them to the input that has the error. Instead, the next tab takes them to the "Clear filters" button. (Note that subsequent presses of the "Apply filters" button aren't announced. That's addressed in another ticket.)

<details><summary>Link, screenshot or steps to recreate</summary>

The video included here should hopefully illustrate all of the cases described above.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/6844775/bac2c752-cefa-4485-a267-24ddfe7ac4a6

</details>

## Recommended action
On most VA.gov forms, focus moves to the containing `div` around the `label`, error message, and `input`, using `tabindex="-1" to assign focus to the non-interactive `div`. This ensures that the `input` will be the next tab stop after hearing the error message.

Note also that the error message should be associated with the `input` using `aria-describedby`. This is not the case as currently coded.

## References

- **WCAG Success Criteria:** [WCAG 2.4.3 Focus Order - Level A](https://www.w3.org/TR/WCAG21/#focus-order)
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** Platform OCTO leadership asks that we document all issues we identify. If you believe that this issue is out of scope or not your team's responsibility, it is still your responsibility to resolve the issue. Please comment on Slack in your team channel tagging `@platform-governance-team-members`, providing an explanation and who you believe is responsible. Governance team will research and follow up.
- [x] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by arianaadili** [comment]
Our team's testing engineer has confirmed the effectiveness of the modifications made to address these concerns. Enclosed are screenshots from our front-end dev team, illustrating how screen reader technology successfully detects the selected content.
Note: when clearing filters, there is a small delay between clearing filters and focus element jumping to 'schools'. The user can tab through the regular flow of elements and once this delay is complete, the focus jumps back to 'school'. 

<img width="827" alt="image-2024-03-04-23-20-11-113" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/98c800b6-2413-4a74-8856-a0e1e9de5bf8">
<img width="641" alt="image-2024-03-04-23-24-02-840" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/d975ad69-75d1-4fd8-ba7d-c37f2e38b294">
<img width="650" alt="image-2024-03-04-23-25-32-308" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/08f4c677-2969-4692-8f8a-2533d85f68a9">
<img width="1075" alt="image-2024-03-04-23-27-20-019" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/fd9b7205-ec1b-4d2d-9d2e-b32da3717ba2">

**Comment by arianaadili** [comment]
Our team's testing engineer has confirmed the effectiveness of the modifications made to address these concerns. Enclosed are screenshots from our front-end dev team, illustrating how screen reader technology successfully detects the selected content.
Note: when clearing filters, there is a small delay between clearing filters and focus element jumping to 'schools'. The user can tab through the regular flow of elements and once this delay is complete, the focus jumps back to 'school'.

<img width="827" alt="310628522-98c800b6-2413-4a74-8856-a0e1e9de5bf8" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/bab58412-16ce-4fcd-a9f9-81c5d1157548">
<img width="641" alt="310628530-d975ad69-75d1-4fd8-ba7d-c37f2e38b294" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/323df795-fcef-4ece-81c5-7462b762e690">
<img width="650" alt="310628537-08f4c677-2969-4692-8f8a-2533d85f68a9 (1)" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/cb8656f6-6518-4dda-9566-0fb348a76371">
<img width="650" alt="310628537-08f4c677-2969-4692-8f8a-2533d85f68a9" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/d293deac-8636-4b56-8fc0-30efa9dc8f4c">
<img width="1075" alt="310628543-fd9b7205-ec1b-4d2d-9d2e-b32da3717ba2" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/b61a1e62-fad3-407b-90d9-2f54e8531b0b">
